test/Waives.Http.Tests/LoggingRequestSenderFacts.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using Waives.Http.Logging;
+using Xunit;
+
+namespace Waives.Http.Tests
+{
+    public class LoggingRequestSenderFacts
+    {
+        private readonly IHttpRequestSender _sender;
+        private readonly ILogger _logger;
+        private readonly LoggingRequestSender _sut;
+        private readonly HttpRequestMessage _request;
+
+        public LoggingRequestSenderFacts()
+        {
+            _sender = Substitute.For<IHttpRequestSender>();
+            _logger = Substitute.For<ILogger>();
+            _sut = new LoggingRequestSender(_sender, _logger);
+
+            _request = new HttpRequestMessage(HttpMethod.Get, new Uri("/documents", UriKind.Relative));
+        }
+
+        [Fact]
+        public async Task Returns_response_from_wrapped_sender()
+        {
+            var expectedResponse = Responses.Success();
+            _sender
+                .Send(Arg.Any<HttpRequestMessage>())
+                .Returns(expectedResponse);
+
+            var response = await _sut.Send(_request);
+
+            Assert.Same(expectedResponse, response);
+        }
+
+        [Fact]
+        public async Task Rethrows_original_exception_when_wrapped_sender_throws_exception()
+        {
+            var expectedException = new WaivesApiException();
+            _sender
+                .Send(Arg.Any<HttpRequestMessage>())
+                .Throws(expectedException);
+
+            try
+            {
+                await _sut.Send(_request);
+            }
+            catch (WaivesApiException e)
+            {
+                Assert.Same(expectedException, e);
+                return;
+            }
+
+            Assert.False(true);
```

Comment:
💪 It would be worth passing a failure message to this assertion which will be displayed in place of the usual "expected: false, received: true". Alternatively, rewriting it with `Assert.Throws()` would make it much clearer, I think.